gammapy/obs/observation.py

```diff
@@ -319,3 +339,569 @@ def select_observations(self, selection=None):
+                s += str(self.obs_groups_table[self.obs_group_axes[i_axis].name][row])
+            s += ' ' + str(self.obs_group_axes[i_axis].bins.unit)
+        return s
+
+    def print_axes(self):
+        """Print axes info using the logging module."""
+        #print(self.info)
+        log.info(self.info)
+
+    def print_groups(self):
+        """Print groups info using the logging module."""
+        #print(self.obs_groups_table)
+        log.info(print(self.obs_groups_table))
+
+    def group_observation_table(self, obs_table):
+        """
+        Group observations in a list according to the defined groups.
+
+        The method returns the same observation table with an extra
+        column in the 1st position indicating the group ID of each
+        observation.
+
+        The algorithm expects the same format (naming and variable
+        definition range) for both the grouping axis definition and
+        the corresponding variable in the table. For instance, if the
+        azimuth axis binning is defined as ``AZ`` with bin edges
+        ``[-90, 90, 270]`` (North and South bins), the input obs table
+        should have an azimuth column defined as ``AZ`` and wrapped
+        at ``270 deg``. This can easily be done by calling:
+
+        >>> obs_table['AZ'] = Angle(obs_table['AZ']).wrap_at(Angle(270., 'degree'))
+
+        Parameters
+        ----------
+        obs_table : `~gammapy.obs.ObservationTable`
+            Observation list to group.
+
+        Returns
+        -------
+        obs_table_grouped : `~gammapy.obs.ObservationTable`
+            Grouped observation list.
+        """
+        if 'GROUP_ID' in obs_table.colnames:
+            raise KeyError(
+                "Catched attempt to overwrite existing grouping in the table.")
+
+        # read the obs groups table row by row (i.e. 1 group at
+        # a time) and lookup the range/value for each parameter
+        n_axes = len(self.obs_group_axes)
+        list_obs_table_grouped = []
+        for i_row in np.arange(self.n_groups):
+            i_group = self.obs_groups_table['GROUP_ID'][i_row]
+            # loop over obs group axes to find out the names and formats
+            # of the parameters to define the selection criteria
+            obs_table_selected = obs_table
+            for i_axis in np.arange(n_axes):
+                name = self.obs_group_axes[i_axis].name
+                format = self.obs_group_axes[i_axis].format
+
+                if format == 'bin_edges':
+                    min_value = recover_units(self.obs_groups_table[name + '_MIN'][i_row],
+                                              self.obs_groups_table[name + '_MIN'])
+                    max_value = recover_units(self.obs_groups_table[name + '_MAX'][i_row],
+                                              self.obs_groups_table[name + '_MAX'])
+                elif format == 'bin_values':
+                   min_value = recover_units(self.obs_groups_table[name][i_row],
```

Comment:
Use 4-space indentation.
(there's a few other PEP8 violations in your code ... it would be nice if you could run `pep8` and / or `pylint` and clean those up yourself)
(the 80 character line length limit doesn't have to be followed strictly ... if the code is more readable and has 85 lines, keep as is)